lib/ddtrace/tracer.rb

```diff
@@ -327,24 +329,14 @@ def trace(name, options = {})
     def record(context)
       context = context.context if context.is_a?(Datadog::Span)
       return if context.nil?
-      trace, sampled = context.get
-
-      # If context flushing is configured...
-      if @context_flush
-        if sampled
-          if trace.nil? || trace.empty?
-            @context_flush.each_partial_trace(context) do |t|
-              write(t)
-            end
-          else
-            write(trace)
-          end
-        end
-      # Default behavior
-      else
-        ready = !trace.nil? && !trace.empty? && sampled
-        write(trace) if ready
-      end
+
+      record_context(context)
+    end
+
+    def record_context(context)
+      trace = @context_flush.consume(context)
```

Comment:
A comment here might be nice, like:

```ruby
# ``#consume`` will return a list of spans if the ``Context`` has spans to record or ``nil`` if there are no spans to record or the trace was not sampled
```

That is kind of lengthy, probably better way to write it, but I think knowing here that `consume(context)` could return no trace if it wasn't sampled could be nice for future people.